### PR TITLE
Added utf8bytes, a wrapper around bytes to ease conversion from bytes to runes

### PR DESCRIPTION
**Pull request by camsn0w**
This will ease the use of undo.Buffer and moving the codebase to use utf8 strings over runes. #334

### CODE REVIEW

File: internal/utf8Bytes/bytes.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		}
+	}
+	// ASCII is simple.  Also, the empty string is ASCII.
+	bytes.numRunes = len(contents)
+	bytes.width = 0
+	bytes.nonASCII = len(contents)
+	return bytes
+}
+
+// Bytes returns the contents of the Bytes.  This method also means the
```
Comment (paul-lalonde) [inline internal/utf8Bytes/bytes.go:57]: Seems suspicious that the function name disagrees with the comment.

**Review by paul-lalonde — COMMENTED** [review]
This cribbed from golang's string handling?

Need an attribution somewhere pointing directly at that, just so we don't lose that history

**Comment by camsn0w** [comment]
> This cribbed from golang's string handling?
> 
> Need an attribution somewhere pointing directly at that, just so we don't lose that history

It comes from https://pkg.go.dev/golang.org/x/exp/utf8string I included the license, is there a specific way that I should be attributing things or does it depend on the source?

**Comment by paul-lalonde** [comment]
Just a pointer to that package in a comment saying what you did to it.

Thanks,
  Paul

**Review by rjkroege — APPROVED** [review]

**Pull request merged**